src/grt/src/fastroute/src/FastRoute.cpp

```diff
@@ -518,11 +518,19 @@ void FastRouteCore::setLayerOrientation(int x)
   layerOrientation = x;
 }
 
-void FastRouteCore::addPin(int netID, int x, int y, int layer)
+void FastRouteCore::addPin(int netID, int x, int y, int layer, bool isRoot)
 {
-  nets[netID]->pinX.push_back(x);
-  nets[netID]->pinY.push_back(y);
-  nets[netID]->pinL.push_back(layer);
+  if (isRoot) {
+    nets[netID]->pinX.insert(nets[netID]->pinX.begin(), x);
```

Comment:
factor out nets[netID]
at least add a comment that this is moving the root pin to the beginning of the vector. it is super confusing to look at. I am hoping to update pdrev to pass in the root index instead of requiring it to be 0. I think it was you who skipped that part of the interface when it was imported into grt.

Reply:
I use std::swap to move it into position 0 (see rsz/src/SteinerTree.cc).
That is way more efficient than moving all the entries up.